### PR TITLE
feat: user pages

### PR DESCRIPTION
**Pull request by brunocorrea23**
https://github.com/denoland/saaskit/issues/181

### CODE REVIEW

File: routes/user/[username].tsx [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+    const user = await getUserByLogin(username);
+    if (user === null) {
+      return ctx.renderNotFound();
+    }
+
+    return ctx.render({
+      ...ctx.state,
+      user: user!,
+    });
```
Comment (iuioiua) [inline routes/user/[username].tsx:26]:
```suggestion
    return ctx.render({ ...ctx.state, user });
```

File: routes/user/[username].tsx [hunk 2]
```diff
@@ -0,0 +1,90 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import type { Handlers, PageProps } from "$fresh/server.ts";
+import Head from "@/components/Head.tsx";
+import Layout from "@/components/Layout.tsx";
+import { BUTTON_STYLES, NOTICE_STYLES } from "@/utils/constants.ts";
```
Comment (iuioiua) [inline routes/user/[username].tsx:5]:
```suggestion
```

File: routes/user/[username].tsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+        )}
+      </div>
+    </li>
+  );
+}
+
+export default function UserPage(props: PageProps<UserData>) {
+  return (
+    <>
+      <Head title="User" href={props.url.href} />
```
Comment (iuioiua) [inline routes/user/[username].tsx:69]:
```suggestion
      <Head title={props.data.user.login} href={props.url.href} />
```

**Review by iuioiua — CHANGES_REQUESTED** [review]
You're heading in the right direction! However, the page I had in mind had an avatar image and username at the top, aligned vertically or horizontally (whichever looks and feels nicer). The rest of the page would be a feed of the items by that user, using a new `getItemsByUser()` in `/utils/db.ts` and using `<ItemSummary />`.

Please let me know if you need any help with anything.

**Comment by brunocorrea23** [comment]
nvm, already included the new getItemsByUser() in /utils/db.ts and also added the items to the user page

**Review by iuioiua — APPROVED** [review]
This is 🔥

We can de-duplicate some of the shared code between the user page and the home page in a follow-up PR. Thank you, @brunocorrea23.

**Pull request merged**